### PR TITLE
Make user SMTP config more priority than server

### LINKED ISSUE #3552: Email Server-wide Vs Per-user configuration
I've noticed that when server-wide email configuration is done then Per-user don't work. Traccar always sends email using the default, not the user configured SMTP server. Is that the desired behavor?

A user might want to use their own SMTP server, not the default server-wide. Is that possible to change this behavior

### PR DESCRIPTION
**Pull request by Abyss777**
Made user SMTP config more priority and added switch to ignore user config

fix #3552

### CODE REVIEW

File: src/org/traccar/notification/NotificationMail.java [hunk 1]
```diff
@@ -87,9 +87,9 @@ private static Properties getProperties(PropertiesProvider provider) {
     public static void sendMailSync(long userId, Event event, Position position) throws MessagingException {
         User user = Context.getPermissionsManager().getUser(userId);
 
-        Properties properties = getProperties(new PropertiesProvider(Context.getConfig()));
-        if (!properties.containsKey("mail.smtp.host")) {
-            properties = getProperties(new PropertiesProvider(user));
+        Properties properties = getProperties(new PropertiesProvider(user));
+        if (Context.getConfig().getBoolean("mail.smtp.ignoreUserConfig") || !properties.containsKey("mail.smtp.host")) {
```
Comment (tananaev) [inline src/org/traccar/notification/NotificationMail.java]: Seems a bit inefficient to initialise user properties if `ignoreUserConfig` is set to `true`.

Comment (Abyss777) [reply src/org/traccar/notification/NotificationMail.java]: I also noticed it, but it become look more complex
```java
        Properties properties;
        if (!Context.getConfig().getBoolean("mail.smtp.ignoreUserConfig")) {
            properties = getProperties(new PropertiesProvider(user));
            if (!properties.containsKey("mail.smtp.host")) {
                properties = getProperties(new PropertiesProvider(Context.getConfig()));
            }
        } else {
            properties = getProperties(new PropertiesProvider(Context.getConfig()));
        }
        if (!properties.containsKey("mail.smtp.host")) {
            Log.warning("No SMTP configuration found");
            return;
        }
```
is it OK? or you see how make simpler?

Comment (tananaev) [reply src/org/traccar/notification/NotificationMail.java]: How about something like this:
```java
Properties properties = null;
if (!Context.getConfig().getBoolean("mail.smtp.ignoreUserConfig")) {
    properties = getProperties(new PropertiesProvider(user));
}
if (properties == null || !properties.containsKey("mail.smtp.host")) {
    properties = getProperties(new PropertiesProvider(Context.getConfig()));
}
```

**Comment by Abyss777** [comment]
Of course! Thanks for advice. Replaced the commit.

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**